debugConsole.go

```diff
@@ -27,110 +27,48 @@ func AddCommand(s string, fn func([]string)) {
 	commands[s] = fn
 }
 
-func debugConsole(resetCh, skipScene chan bool) {
-	scanner := bufio.NewScanner(os.Stdin)
+func debugConsole(resetCh, skipScene chan bool, input io.Reader) {
+	scanner := bufio.NewScanner(input)
 	spew.Config.DisableMethods = true
 	spew.Config.MaxDepth = 2
+
+	builtinCommands = map[string]func([]string){
+		"viewport": viewportCommands,
+		"fade":     fadeCommands,
+		"skip":     skipCommands(skipScene),
+		"print":    printCommands,
+		"mouse":    mouseCommands,
+	}
+
 	for {
 		select {
 		case <-resetCh: //reset all vars in debug console that save state
 			viewportLocked = true
 		default:
 		}
 		for scanner.Scan() {
-			select {
-			case <-resetCh: //reset all vars in debug console that save state
-				viewportLocked = true
-			default:
-			}
 			//Parse the Input
 			tokenString := strings.Fields(scanner.Text())
-			if len(tokenString) == 0 {
+			if len(tokenString) < 2 {
 				continue
 			}
 
-			// These different commands should probably be split off, so that
+			// The builtin commands should probably be split off, so that
```

Comment:
There probably should be some more thought into what other types of general commands a user might desire. That being said it makes sense to split off the built in and perhaps give an easy way to toggle either a single command type or all built-in commands.